### PR TITLE
Replace custom job names with namespaces

### PR DESCRIPTION
**Pull request by AndrewChubatiuk**
First of all I'd like to say thank you for your project. It's amazing and the closest from all existing ones to what I need. I prepared a PR with changes, that I need to support exporting of metrics from services, that we want to monitor in our project. 
- Replaced custom service names to Cloudwatch namespace names which helped to have a single dictionary with all supported services instead on multiple
- Added regular expressions to parse dimensions from resources
- Removed a need to use static dimensions in dynamic jobs in cases, when they cannot be parsed from ARNs

### CODE REVIEW

**Comment by thomaspeitz** [comment]
Wow that looks incredible but it is so big. So could take some time to review. Appreciate your work!

**Comment by thomaspeitz** [comment]
Could you please rebase? :)

I read over the code and it looks good, I would like to check it out locally and test it first before merging.

Would like to merge and release it the following seven days.

Greetings, Thomas :)

**Comment by AndrewChubatiuk** [comment]
@thomaspeitz 
Thank you. Thomas:)
I've resolved conflicts

**Comment by thomaspeitz** [comment]
Awesome. Thanks a lot for your work! - Totally appreciate the effort.

**Comment by thomaspeitz** [comment]
So I made a quick review. I am unhappy about "AWS/ApplicationELB" settings. I would like to keep a custom naming.
I see that a naming e.g. AWS/ApplicationELB is the right one in terms of API but I do not like it terms of user experience.

For me it is more important the exporter has a nice usability than a nice developer feeling. So I would like to suggest that we keep a mapping between our names and AWS API names. I would like to have same "guessable" which is hard with aws naming schema - That would be a start to discuss the naming. I tried to keep them simple and without casing. So they are easier to guess:
```
elb-application - AWS/ApplicationELB - Application Load Balancer
apigateway - AWS/ApiGateway - Api Gateway
appsync - AWS/AppSync - AppSync
billing - AWS/Billing - Billing
cloudfront - AWS/CloudFront - Cloud Front
docdb - AWS/DocDB - DocumentDB (with MongoDB compatibility)
dynamodb - AWS/DynamoDB - NoSQL Online Datenbank Service
ebs - AWS/EBS - Elastic Block Storage
elasticache - AWS/Elasticache - ElastiCache
ec2 - AWS/EC2 - Elastic Compute Cloud
ec2-spot - AWS/EC2Spot - Elastic Compute Cloud for Spot Instances
ecs - AWS/ECS - Elastic Container Service (Service Metrics)
ecs-containerinsights -ECS/ContainerInsights - ECS/ContainerInsights (Fargate metrics)
efs- AWS/EFS - Elastic File System
elb - AWS/ELB - Elastic Load Balancer
elasticmapreduce - AWS/ElasticMapReduce - Elastic MapReduce
elasticsearch - AWS/ES - ElasticSearch
fsx - AWS/FSx - FSx File System
gamelift - AWS/GameLift - GameLift
glue - Glue - AWS Glue Jobs
kinesis -AWS/Kinesis - Kinesis Data Stream
natgateway -AWS/NATGateway - Nat Gateway
lambda - AWS/Lambda - Lambda Functions
elb-network -AWS/NetworkELB - Network Load Balancer
redshift - AWS/Redshift - Redshift Database
rds - AWS/RDS - Relational Database Service
route53-resolver - AWS/Route53Resolver - Route53 Resolver
s3 - AWS/S3 - Object Storage
sqs - AWS/SQS - Simple Queue Service
transitgateway - AWS/TransitGateway - Transit Gateway
vpn - AWS/VPN - VPN connection
autoscaling - AWS/AutoScaling - Auto Scaling Group
kafka - AWS/Kafka - Managed Apache Kafka
firehose - AWS/Firehose - Managed Streaming Service
sns - AWS/SNS - Simple Notification Service
stepfunctions - AWS/States - Step Functions
waf - AWS/WAFV2 - Web Application Firewall v2
```

How is your opinion about guessable service names?

**Comment by AndrewChubatiuk** [comment]
@thomaspeitz 
I've added support of both aliases, that you've listed in a comment above and namespaces

**Comment by thomaspeitz** [comment]
@AndrewChubatiuk - Lets please make the old ones (alias) the default ones. - Especially in docs / Readme.md. Thanks. Beside of that I would merge it and try it out.

**Comment by thomaspeitz** [comment]
Looks awesome. Thanks a lot!

File: README.md [hunk 1]
```diff
@@ -151,17 +152,17 @@ general setting.  The currently inherited settings are period, and addCloudwatch
 ```yaml
 discovery:
   exportedTagsOnMetrics:
-    ec2:
+    AWS/EC2:
       - Name
-    ebs:
+    AWS/EBS:
```
Comment (thomaspeitz) [inline README.md:157]: Can we change these one, please? :)

File: README.md [hunk 2]
```diff
@@ -206,30 +205,30 @@ discovery:
         length: 900 #(this will be ignored)
         delay: 300 #(this will be ignored)
         nilToZero: true
-  - type: alb
+  - type: AWS/ApplicationELB
```
Comment (thomaspeitz) [inline README.md:208]: Same

File: config_test.yml [hunk 3]
```diff
@@ -1,44 +1,51 @@
...
   exportedTagsOnMetrics:
-    ec2:
-      - Name
-    ebs:
+    AWS/EBS:
       - VolumeId
+    kafka:
+      - Name
   jobs:
+  - type: AWS/Billing
```
Comment (thomaspeitz) [inline config_test.yml:8]: Lets fix this one as well.

File: main.go [hunk 4]
```diff
@@ -133,7 +93,8 @@ func main() {
 		length := getMetricDataInputLength(discoveryJob)
 		//S3 can have upto 1 day to day will need to address it in seprate block
```
Comment (thomaspeitz) [inline main.go:94]: Should we remove the command as you seem to have fixed the s3 problems?

**Review by thomaspeitz — CHANGES_REQUESTED** [review]
Found some small things on second glance

File: main.go (hunk 4 above)
Comment (AndrewChubatiuk) [reply main.go:94]: I haven't fixed it. I've just added a boolean parameter `IgnoreLength` to a struct, which says that a lenght of this metric can be ignored, cause now not only S3 metrics are updated seldom but also Billing ones

Comment (thomaspeitz) [reply main.go:94]: Thanks for the explanation.

**Pull request merged**